### PR TITLE
Parsing Travis logs to extract knapsack's json

### PR DESCRIPTION
**Pull request by dquimper**
* Parsing Travis logs to extract knapsack's JSON
* Create a pull request with a renewed version.

### CODE REVIEW

File: bin/travis_analyzer [hunk 1]
```diff
@@ -13,6 +17,7 @@ class TravisAnalyzer
 
     @upgrade_analyzer = UpgradeAnalyzer::Analyzer.new(@travis, repo_name, github_token)
     @log_analyzer = LogAnalyzer::Analyzer.new(@travis, repo_name, github_token)
+    @knapsack_analyzer = KnapsackAnalyzer::Analyzer.new(@travis, repo_name, github_token)
```
Comment (zefta) [inline bin/travis_analyzer:20]: I think this should not be in this commit since the class doesn't exist yet.

**Comment by zefta** [comment]
Overall it looks good to me.

**Comment by zefta** [comment]
Autre chose le rapport qui est generer pourrait afficher plus que les 3 plus lent peut etre les 10 premiers ou quelque chose comme ca?

File: lib/log_analyzer/analyzer.rb [hunk 2]
```diff
@@ -40,7 +40,7 @@ def report_results(build)
     end
 
     def log(msg)
-      puts msg
+      puts "LogAnalyzer: #{msg}"
```
Comment (loto) [inline lib/log_analyzer/analyzer.rb]: What about using a logger? So we can manage more options: decide where to output the logs, level of detail, etc.

File: lib/upgrade_analyzer/analyzer.rb [hunk 3]
```diff
@@ -146,7 +146,7 @@ def report_invalid_comparison(github, errors)
     end
 
     def log(msg)
-      puts msg
+      puts "UpgradeAnalyzer: #{msg}"
```
Comment (loto) [inline lib/upgrade_analyzer/analyzer.rb]: What about using a logger? So we can manage more options: decide where to output the logs, level of detail, etc.

**Review by loto — COMMENTED** [review]
Nice work!

File: bin/travis_analyzer (hunk 1 above)
Comment (dquimper) [reply bin/travis_analyzer:20]: fixed

File: lib/log_analyzer/analyzer.rb (hunk 2 above)
Comment (dquimper) [reply lib/log_analyzer/analyzer.rb]: Done!

File: lib/upgrade_analyzer/analyzer.rb (hunk 3 above)
Comment (dquimper) [reply lib/upgrade_analyzer/analyzer.rb]: Done!

**Review by Raindal — APPROVED** [review]
Let's see where it goes from here

**Review by zefta — APPROVED** [review]

**Pull request merged**